src/integration-test/kotlin/integration/btc/BtcPreGenIntegrationTest.kt

```diff
@@ -0,0 +1,90 @@
+package integration.btc
+
+import enums.Domain
+import integration.helper.IntegrationHelperUtil
+import kotlinx.coroutines.experimental.async
+import mu.KLogging
+import org.bitcoinj.core.ECKey
+import org.bitcoinj.core.Utils
+import org.bitcoinj.params.RegTestParams
+import org.bitcoinj.script.ScriptBuilder
+import org.bitcoinj.wallet.Wallet
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.fail
+import provider.TriggerProvider
+import provider.btc.BtcSessionProvider
+import registration.btc.pregen.executePreGeneration
+import util.getRandomString
+import java.io.File
+
+class BtcPreGenIntegrationTest {
+
+    private val integrationHelper = IntegrationHelperUtil()
+
+    private val btcPkPreGenConfig =
+        integrationHelper.configHelper.createBtcPreGenConfig()
+
+    private val triggerProvider = TriggerProvider(
+        btcPkPreGenConfig.iroha,
+        btcPkPreGenConfig.pubKeyTriggerAccount,
+        btcPkPreGenConfig.registrationAccount
+    )
+    private val btcKeyGenSessionProvider = BtcSessionProvider(
+        btcPkPreGenConfig.iroha,
+        btcPkPreGenConfig.registrationAccount,
+        integrationHelper.irohaKeyPair
+    )
+
+    /**
+     * Test US-001 btc addresses pregeneration
+     * Note: Iroha must be deployed to pass the test.
+     * @given session account is created
+     * @when special trigger account is triggered
+     * @then new multisig btc address is created
+     */
+    @Test
+    fun testGenerateKey() {
+        integrationHelper.registerNotary("test", 8080)
+        async { executePreGeneration(btcPkPreGenConfig) }
+        Thread.sleep(10_000)
+        val sessionAccountName = String.getRandomString(9)
```

Comment:
Add some prefix

Reply:
We can easily distinguish session accounts from the others. It has it's own domain called "btcSession". I don't think we need it.